src/rbdmap-generator

```diff
@@ -0,0 +1,299 @@
+{unit_extra}
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/bin/rbd device map %I{rbd_map_args}
+ExecStartPost=-/etc/ceph/rbd.d/%I map /dev/rbd/%I
+ExecStop=-/etc/ceph/rbd.d/%I unmap /dev/rbd/%I
+ExecStop=/usr/bin/rbd device unmap %I{rbd_unmap_args}
+'''
+
+VALID_CHARS = string.ascii_letters + string.digits + ':_.'
+
+def systemd_escape(p: str, path=True):
+    res = ''
+    for c in p:
+        if c == '/':
+            res += '-'
+        elif c in VALID_CHARS:
+            res += c
+        else:
+            res += ''.join(R'\x{:02x}'.format(b) for b in c.encode())
+    if res[0] == '.':
+        res = R'\x2e' + res[1:]
+    return res
+
+def systemd_escape_path(p: str):
+    if (p == '/'):
+        return '-'
+    p = p.strip('/')
+    p = re.sub(r'/{2,}', '/', p)
+    return systemd_escape(p)
+
+def normalize_rbd_spec(spec: str):
+    if spec.startswith('/') or spec.endswith('/'):
+        raise ValueError(
+            'rbd image or snapshot specification {} cannot starts or ends with /'
+            .format(spec))
+    if '/' not in spec:
+        spec = 'rbd/' + spec
+    return spec
+
+def split_rbdmap_options(options: str):
+    in_quote = False
+    working = ''
+    for c in options:
+        if c == "'":
+            in_quote = not in_quote
+        elif c == ',' and not in_quote:
+            yield working
+            working = ''
+        else:
+            working += c
+    yield working
+
+class RbdMapSpec(NamedTuple):
+    rbd_spec: str
+    systemd_unit_options: Dict[str, List[str]]
+    device_type: Optional[str]
+    map_generic_args: List[str]
+    map_options: Optional[str]
+    unmap_options: Optional[str]
+
+def empty_systemd_unit_options():
+    return {
+        'required-by': [],
+        'wanted-by': [],
+        'requires': [],
+        'requires-mounts-for': [],
+        'before': [],
+        'after': [],
+    }
+    
+def parse_rbdmap_line(line: str):
+    cells = line.split(maxsplit=1)
+    rbd_spec = cells[0]
+    opts = cells[1] if len(cells) >= 2 else None
+    map_generic_args = []
+    device_type = None
+    map_options = None
+    if opts:
+        for a in split_rbdmap_options(opts):
+            kv = a.split('=', maxsplit=1)
+            if kv[0] == 'device-type':
+                if len(kv) < 2:
+                    raise ValueError('device-type should have a value')
+                device_type = kv[1]
+            elif kv[0] == 'options':
+                if len(kv) >= 2:
+                    map_options = kv[1]
+            else:
+                map_generic_args.append(a)
+
+    return RbdMapSpec(rbd_spec, empty_systemd_unit_options(), device_type, map_generic_args, map_options, None)
+
+SYSTEMD_KEY_PREFIX = 'x-systemd.'
+RBDMAP_TARGET = 'rbdmap.target'
+
+def parse_rbdtab_line(line: str):
+    bool_options = {
+        'fail': True,
+        'auto': True,
+    }
+    systemd_unit_options = empty_systemd_unit_options()
+
+    device_type, rbd_spec, *cells = line.split(maxsplit=4)
```

Comment:
May be not use this `maxsplit=4` limitation, just in case we want to add a new field in the future and still have the old scripts to work (ignoring it)?